api/router.mjs

```diff
@@ -0,0 +1,42 @@
+import { NOT_FOUND } from './defs.mjs'
+import * as db from './db.mjs'
+
+const handlers = {}
+const setter = (p) => (_, k, v) => (handlers[`${p}/${k}`] = v)
+const getter = (p) => (cache, k) => cache[k] || (cache[k] = ROUTE(`${p}/${k}`))
+const ROUTE = (p) => new Proxy({}, { get: getter(p), set: setter(p) })
+
+export const GET = ROUTE('GET/api')
+export const POST = ROUTE('POST/api')
+
+export const withUser = (fn) => async (params) => {
+  if (!params.session) return new Response('No Session', UNAUTHORIZED)
+  params.user = await db.get(params.session)
+  if (!params.user) return new Response('Bad Session', UNAUTHORIZED)
+  // ...maybe remove session cookie in that case
+  return fn(params)
+}
+
+export const withRoom = (fn) => async (params) => {
+  const { ROOM } = params.env
+  ROOM.idFromName('nan')
+  params.room = await ROOM.get(id)
+  return fn(params)
+}
+
+export const getCookie = (request, key) => {
+  const cookieStr = request.headers.get('Cookie')
+  if (!cookieStr) return undefined
+  const x = cookieStr.indexOf(`${key}=`)
+  if (x < 0) return
+  const y = cookieStr.indexOf('; ', x)
+  return cookieStr.slice(x + key.length + 1, y < x ? cookieStr.length : y)
+}
+
+export const handleRequest = async (request, env) => {
+  const url = new URL(request.url)
+  const handler = handlers[`${request.method}${url.pathname}`]
+  if (!handler) return new Response(null, NOT_FOUND)
+  const session = getCookie(request, 'nan-session')
+  return handler({ request, url, session, env })
```

Comment:
This is the main handler:
- first I parse the url, to get the `pathname` without eventual `?seach=params`
- then, I look if I have a handler set for this path
- if not -> 404 otherwhise I get the session and forward the params to the `handler` we matched